api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -109,8 +112,15 @@ public Obs saveObs(Obs obs, String changeMessage) throws APIException {
 			setPersonFromEncounter(obs);
 			return saveObsNotDirty(obs, changeMessage);
 		} else {
-			setPersonFromEncounter(obs);
-			return saveExistingObs(obs,changeMessage);
+			Errors errors = new BindException(obs, "obs");
```

Comment:
You don't need any changes in this class, all the validation needs to go to the ObsValidator class, you won't need to instantiate BindException there